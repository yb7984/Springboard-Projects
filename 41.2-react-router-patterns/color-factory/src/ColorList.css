.ColorList{
    background-color: blanchedalmond;
    width: 100vw;
    height: 100vh;
}

.ColorList-header{
    padding: 50px;
    background-color: black;
    color: white;
}

.ColorList-colors{
    font-size: 1.5em;
}