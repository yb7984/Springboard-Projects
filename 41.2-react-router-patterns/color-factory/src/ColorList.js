import { Link } from 'react-router-dom';
import './ColorList.css';

const ColorList = ({ colors = [] }) => (
    <div className="ColorList">
        <div className="ColorList-header">
            <h1>Welcome to the color factory</h1>
            <h2><Link to='/colors/new'>Add a color</Link></h2>
        </div>
        <div>
            <h1>Please select a color:</h1>
            
            {
                colors.map(color => (
                    <div key={color.name} className="ColorList-colors">
                        <Link to={`/colors/${color.name}`}>{color.name}</Link>
                    </div>
                ))}
        </div>
    </div>
)

export default ColorList;