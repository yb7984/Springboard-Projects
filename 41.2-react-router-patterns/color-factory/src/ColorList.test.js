import ColorList from './ColorList';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';

const colors = [
    {
        name: "Red",
        value: "#FF0000"
    }
];


it('renders without crashing', () => {
    render(<MemoryRouter><ColorList colors={colors} /></MemoryRouter>);
});


it("matches snapshot", function () {
    const { asFragment, queryByText } = render(
        <MemoryRouter>
            <ColorList colors={colors} />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(queryByText('Add a color')).toBeInTheDocument();
    expect(queryByText('Red')).toBeInTheDocument();
});
