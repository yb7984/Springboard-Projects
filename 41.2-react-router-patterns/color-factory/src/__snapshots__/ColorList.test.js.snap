// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ColorList"
  >
    <div
      class="ColorList-header"
    >
      <h1>
        Welcome to the color factory
      </h1>
      <h2>
        <a
          href="/colors/new"
        >
          Add a color
        </a>
      </h2>
    </div>
    <div>
      <h1>
        Please select a color:
      </h1>
      <div
        class="ColorList-colors"
      >
        <a
          href="/colors/Red"
        >
          Red
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, works with the colorName in colors 1`] = `
<DocumentFragment>
  <div
    class="ColorList"
  >
    <div
      class="ColorList-header"
    >
      <h1>
        Welcome to the color factory
      </h1>
      <h2>
        <a
          href="/colors/new"
        >
          Add a color
        </a>
      </h2>
    </div>
    <div>
      <h1>
        Please select a color:
      </h1>
      <div
        class="ColorList-colors"
      >
        <a
          href="/colors/Red"
        >
          Red
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
